prusti-viper/src/encoder/mir/procedures/encoder/mod.rs

```diff
@@ -821,6 +821,44 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         Ok(())
     }
 
+    fn encode_close_mut_ref(
+        &mut self,
+        block_builder: &mut BasicBlockBuilder,
+        location: mir::Location,
+        base: &vir_high::Expression,
+        object: vir_high::Expression,
+    ) -> SpannedEncodingResult<()> {
+        if let vir_high::ty::Type::Reference(vir_high::ty::Reference { lifetime, .. }) =
```

Comment:
If I understand correctly, this is called only when `base` is known to be a reference, right? If yes, I would suggest adding `else { unreachable!() }` to this if statement.

Reply:
that is correct, done